src/main/java/io/xpring/xrpl/model/XRPIssuedCurrency.java

```diff
@@ -0,0 +1,46 @@
+package io.xpring.xrpl.model;
+
+import org.xrpl.rpc.v1.IssuedCurrencyAmount;
+import java.math.BigInteger;
+import org.immutables.value.Value;
+
+/**
+ * An issued currency on the XRP Ledger
+ * @see "https://xrpl.org/basic-data-types.html#specifying-currency-amounts"
+ */
+@Value.Immutable
+public interface XRPIssuedCurrency {
+    static ImmutableXRPIssuedCurrency.Builder builder() {
+        return ImmutableXRPIssuedCurrency.builder();
+    }
+
+    /**
+     * @return The currency used to value the amount.
+     */
+    XRPCurrency currency();
+
+    /**
+     * @return The value of the amount.
+     */
+    BigInteger value();
+
+    /**
+     * @return Unique account address of the entity issuing the currency.
+     */
+    String issuer();
+
+    static XRPIssuedCurrency from(IssuedCurrencyAmount issuedCurrency) {
+        BigInteger value;
+        try {
+            value = new BigInteger((issuedCurrency.getValue()));
+        } catch (NumberFormatException e) {
+            return null;
```

Comment:
You might want to just throw the `NumberFormatException` here.  This will likely only happen if there is something wrong on the back end (a bad number is getting passed back), which you will want to know about

Reply:
good idea, will do this